### DISCUSSION TITLE
ImportError: cannot import name 'SwinUNETR'

### DISCUSSION THREAD

**Discussion by Guoxt** [Project-MONAI/MONAI#5704] [Q&A]
from monai.networks.nets import SwinUNETR
ImportError: cannot import name 'SwinUNETR'

What's the reason？？？

**Comment by KumoLiu** [comment]
Hi @Guoxt, could you please help check your environment, seems you are in the wrong environment. Thanks!

**Reply by Guoxt** [reply]
I tried the following methods, but it didn't work.

git clone https://github.com/Project-MONAI/MONAI.git
cd MONAI/
pip install -r requirements-dev.txt

**Reply by KumoLiu** [reply]
Hi @Guoxt, could you please try to follow the steps here?
https://docs.monai.io/en/stable/installation.html#option-2-editable-installation 
Thanks!

**Reply by Guoxt** [reply]
Thank you for your answer. The following is the running result, but I still don't know how to handle it.

root@q23df07391b64ff0b64ee1e0d3449771-task0-0:/MONAI# python setup.py develop
setup.py with torch 1.10.2+cu102
setup.py:46: UserWarning: extension build skipped: 'Version' object has no attribute 'release'
  warnings.warn(f"extension build skipped: {e}")
Please set environment variable `BUILD_MONAI=1` to enable Cpp/CUDA extension build.
BUILD_MONAI_CPP=False, BUILD_MONAI_CUDA=False, TORCH_VERSION=0.
running develop
running egg_info
creating monai.egg-info
writing monai.egg-info/PKG-INFO
writing dependency_links to monai.egg-info/dependency_links.txt
writing requirements to monai.egg-info/requires.txt
writing top-level names to monai.egg-info/top_level.txt
writing manifest file 'monai.egg-info/SOURCES.txt'
reading manifest file 'monai.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'monai.egg-info/SOURCES.txt'
running build_ext
Creating /usr/local/lib/python3.6/dist-packages/monai.egg-link (link to .)
Adding monai 1.1.0rc1+23.g4f584a9 to easy-install.pth file

Installed /MONAI
Processing dependencies for monai==1.1.0rc1+23.g4f584a9
Searching for numpy==1.19.5
Best match: numpy 1.19.5
Adding numpy 1.19.5 to easy-install.pth file
Installing f2py script to /usr/local/bin
Installing f2py3 script to /usr/local/bin
Installing f2py3.6 script to /usr/local/bin

Using /usr/local/lib/python3.6/dist-packages
Searching for torch==1.10.2
Best match: torch 1.10.2
Adding torch 1.10.2 to easy-install.pth file
Installing convert-caffe2-to-onnx script to /usr/local/bin
Installing convert-onnx-to-caffe2 script to /usr/local/bin
Installing torchrun script to /usr/local/bin

Using /usr/local/lib/python3.6/dist-packages
Searching for typing-extensions==4.1.1
Best match: typing-extensions 4.1.1
Adding typing-extensions 4.1.1 to easy-install.pth file

Using /usr/local/lib/python3.6/dist-packages
Searching for dataclasses==0.8
Best match: dataclasses 0.8
Adding dataclasses 0.8 to easy-install.pth file

Using /usr/local/lib/python3.6/dist-packages
Finished processing dependencies for monai==1.1.0rc1+23.g4f584a9

**Reply by KumoLiu** [reply]
Seems it's ok now, did you still can't import correctly?

**Reply by Guoxt** [reply]
Still cannot import correctly.

root@q23df07391b64ff0b64ee1e0d3449771-task0-0:/# python /code/GUOXUTAO/2023_00/Swinunet/3D/01/main.py
/usr/local/lib/python3.6/dist-packages/sklearn/externals/joblib/externals/cloudpickle/cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
Traceback (most recent call last):
  File "/code/GUOXUTAO/2023_00/Swinunet/3D/01/main.py", line 9, in <module>
    from monai.networks.nets import SwinUNETR
ImportError: cannot import name 'SwinUNETR'
root@q23df07391b64ff0b64ee1e0d3449771-task0-0:/# python /code/GUOXUTAO/2023_00/Swinunet/3D/01/main.py
/usr/local/lib/python3.6/dist-packages/sklearn/externals/joblib/externals/cloudpickle/cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
Traceback (most recent call last):
  File "/code/GUOXUTAO/2023_00/Swinunet/3D/01/main.py", line 9, in <module>
    from monai.networks.nets import SwinUNETR
ImportError: cannot import name 'SwinUNETR'

**Reply by Guoxt** [reply]
UNETR can import correctly.

from monai.networks.nets import UNETR

**Reply by KumoLiu** [reply]
Could you please try `pip show monai` in the terminal to check whether you install it correctly?

**Reply by Guoxt** [reply]
The following is the running result：

root@n444226be21e4784a283cf0fd9202b9b-task0-0:/# pip show monai
Name: monai
Version: 0.8.1
Summary: AI Toolkit for Healthcare Imaging
Home-page: https://monai.io/
Author: MONAI Consortium
Author-email: monai.contact@gmail.com
License: Apache License 2.0
Location: /usr/local/lib/python3.6/dist-packages
Requires: torch, numpy
Required-by:

**Reply by KumoLiu** [reply]
Your MONAI version is still 0.8.1, could you please `pip uninstall monai`, remove all the monai folders and retry the steps [here](https://docs.monai.io/en/stable/installation.html#installing-the-recommended-dependencies)
```
cd MONAI/
python setup.py develop --uninstall

# to further clean up the MONAI/ folder (Bash script)
./runtests.sh --clean
```
Thanks!

**Reply by Guoxt** [reply]
Hello. I uninstall the existing version and reinstall it, but the version of MONAI is still 0.8.1.

**Reply by KumoLiu** [reply]
Did you use `pip uninstall monai`?

**Reply by Guoxt** [reply]
Hi. This is my installation and uninstallation record. According to the prompt, it seems that the maximum version is 0.8.1.

```
root@n444226be21e4784a283cf0fd9202b9b-task0-0:/# pip uninstall monai
Found existing installation: monai 0.8.1
Uninstalling monai-0.8.1:
  Would remove:
    /usr/local/lib/python3.6/dist-packages/monai-0.8.1.dist-info/*
    /usr/local/lib/python3.6/dist-packages/monai/*
Proceed (Y/n)? y
  Successfully uninstalled monai-0.8.1
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
root@n444226be21e4784a283cf0fd9202b9b-task0-0:/# pip show monai
WARNING: Package(s) not found: monai
root@n444226be21e4784a283cf0fd9202b9b-task0-0:/# 
root@n444226be21e4784a283cf0fd9202b9b-task0-0:/# pip uninstall -y monai
WARNING: Skipping monai as it is not installed.
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
root@n444226be21e4784a283cf0fd9202b9b-task0-0:/# pip install monai==1.0.0
ERROR: Could not find a version that satisfies the requirement monai==1.0.0 (from versions: 0.0.1, 0.1.0, 0.2.0, 0.3.0, 0.4.0, 0.5.0, 0.5.1, 0.5.2, 0.5.3, 0.6.0, 0.7.0, 0.8.0, 0.8.1)
ERROR: No matching distribution found for monai==1.0.0
root@n444226be21e4784a283cf0fd9202b9b-task0-0:/# pip install monai       
Collecting monai
  Using cached monai-0.8.1-202202162213-py3-none-any.whl (721 kB)
Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.6/dist-packages (from monai) (1.19.5)
Requirement already satisfied: torch>=1.6 in /usr/local/lib/python3.6/dist-packages (from monai) (1.10.2)
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.6/dist-packages (from torch>=1.6->monai) (4.1.1)
Requirement already satisfied: dataclasses in /usr/local/lib/python3.6/dist-packages (from torch>=1.6->monai) (0.8)
Installing collected packages: monai
Successfully installed monai-0.8.1
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
root@n444226be21e4784a283cf0fd9202b9b-task0-0:/# pip show monai
Name: monai
Version: 0.8.1
Summary: AI Toolkit for Healthcare Imaging
Home-page: https://monai.io/
Author: MONAI Consortium
Author-email: monai.contact@gmail.com
License: Apache License 2.0
Location: /usr/local/lib/python3.6/dist-packages
Requires: numpy, torch
Required-by: 
root@n444226be21e4784a283cf0fd9202b9b-task0-0:/# 
```

**Reply by KumoLiu** [reply]
Hi @Guoxt, seems your python version is too low, could you please update your python version >= 3.7 and try it again?

**Reply by KumoLiu** [reply]
You can find the requirement here, https://github.com/Project-MONAI/MONAI/blob/4f584a945a5deb5622fb09d165fe13f5cf46e3d7/setup.cfg#L19

**Reply by Guoxt** [reply]
Problem solved, thank you for your patience!
